### DISCUSSION TITLE
[Neovim plugin] Request signInInitiate failed with message: unable to get local issuer certificate

### DISCUSSION THREAD

**Discussion by holstvoogd** [community/community#49711] [Copilot Conversations]
### Select Topic Area

Bug

### Body

Hi,

I'm trying to evaluate Copilot for our organisation, but am stuck on the 'unable to get local issuer certificate' error reported many times before wrt broken/enterprisey proxy setups etc. However, I am not behind a proxy and even with copilot_proxy_strict_ssl set to false it fails to do anything. One of my colleagues has been using Copilot from VScode without issue. So I'm thinking this is a issue with the plugin or its dependecies?

This is all the output I get:

```
Error detected while processing function copilot#Command[38]..function copilot#Command[30]..3[13]..copilot#Call[2]..<SNR>94_AgentCall[5]..<SNR>94_RequestAwait:
line    5:
E605: Exception not caught: copilot#agent(-32603): Request signInInitiate failed with message: unable to get local issuer certificate
Press ENTER or type command to continue
```

Is there some debug mode I can enable to get more info on what it is trying to do?

**Comment by ghost** [comment]
Hi there, it sounds like you're experiencing issues with the Copilot plugin for Neovim. You may want to try enabling debug mode to get more information on what's causing the problem. You can do this by setting the `g:copilot_debug` variable to `1` in your Neovim configuration. This should output more detailed information to the console when you try to use the plugin.

If that doesn't help, you may want to reach out to the Copilot team directly for further assistance. They may be able to provide more specific guidance on troubleshooting this issue.

**Comment by danielbisar** [comment]
Are you behind a zscaler or VPN? 

See also https://github.com/community/community/discussions/8866

**Reply by holstvoogd** [reply]
I am not behind any kind of VPN on proxy. Just me, my router and a stranf of fiber ;)
Also, the problem does not occur with vscode. So it's clearly something with the vim plugin and/or my setup of it.

**Comment by moooooser999** [comment]
I faced the same issue here, while I am not behind any proxy or VPN

**Comment by holstvoogd** [comment]
Not going to spend anymore energy on this, GH clearly isn't going to fix anything or even respond.

**Comment by wenjie1991** [comment]
After experiencing this issue for three months, I have finally resolved it.

The problem was caused by a node-related issue. Although I am unsure of the exact reason, reinstalling node resolved the problem.

To determine if you are facing the same issue, follow these steps:

**Step 1: Verify if the problem is occurring with node.**

If the following line works, it indicates that the certification is functioning correctly at the system level.

```
curl -v https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
```

Then check the node, we can create a simplest script to test it.

```
const https = require('https');

https.get('https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex', (res) => {
  console.log('statusCode:', res.statusCode);
  console.log('headers:', res.headers);
}).on('error', (e) => {
  console.error(e);
});
```
Save the above javascript to `script.js` and run it `node script.js`.

If you see the following errors, congulatulation! Try the step2, it may help 
```
Error: unable to get local issuer certificate
      at TLSSocket.onConnectSecure (node:_tls_wrap:1627:34)
      at TLSSocket.emit (node:events:514:28)
      at TLSSocket._finishInit (node:_tls_wrap:1038:8)
      at ssl.onhandshakedone (node:_tls_wrap:824:12) {
    code: 'UNABLE_TO_GET_ISSUER_CERT_LOCALLY'
```
**Step2, re-install the node. For me, the most security way is to use mvn to install a brand new node in user fold.**
```
curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.1/install.sh | bash
nvm install node
```

Good luck!

**Reply by scribu** [reply]
I had the same error, but reinstalling Node didn't work for me.

What worked is adding this environment variable in my .zprofile and restarting the terminal:

```bash
export NODE_EXTRA_CA_CERTS=/private/etc/ssl/cert.pem
```

**Reply by tuan-le-zalora** [reply]
this work for me as well
changed node version to 20 some time ago and forgot it
so I changed the version 22 it and work again
